product_code/lagom/shared/client/src/main/scala/de/upb/cs/uc4/shared/client/CustomExceptionSerializer.scala

```diff
@@ -0,0 +1,126 @@
+package de.upb.cs.uc4.shared.client
+
+import java.io.{CharArrayWriter, PrintWriter}
+
+import akka.util.ByteString
+import com.lightbend.lagom.scaladsl.api.deser.{DefaultExceptionSerializer, RawExceptionMessage}
+import com.lightbend.lagom.scaladsl.api.transport.{ExceptionMessage, MessageProtocol, TransportErrorCode, TransportException}
+import play.api.libs.json.{JsArray, JsError, JsSuccess, Json}
+import play.api.{Environment, Mode}
+
+import scala.util.control.NonFatal
+
+class CustomExceptionSerializer(environment: Environment) extends DefaultExceptionSerializer(environment) {
+
+  override def serialize(exception: Throwable, accept: Seq[MessageProtocol]): RawExceptionMessage = {
+    val (errorCode, message) = exception match {
+      case ce: CustomException => (ce.getErrorCode, ce.getPossibleErrorResponse)
+      case te: TransportException =>
+        (te.errorCode, te.exceptionMessage)
+      case e if environment.mode == Mode.Prod =>
+        // By default, don't give out information about generic exceptions.
+        (TransportErrorCode.InternalServerError, new ExceptionMessage("Exception", ""))
+      case e =>
+        // Ok to give out exception information in dev and test
+        val writer = new CharArrayWriter
+        e.printStackTrace(new PrintWriter(writer))
+        val detail = writer.toString
+        (TransportErrorCode.InternalServerError, new ExceptionMessage(s"${exception.getClass.getName}: ${exception.getMessage}", detail))
+    }
+
+    val messageBytes = message match {
+      case per :DetailedError =>
```

Comment:
Renaming of per to something fitting to DetailedError (de f. e.) should be considered